Asset Setup and Logo Integration in Header

Winnycloud27 opened the pull request:
This pull request addresses the task of setting up the assets directory and integrating the favicon into the header component. The changes include creating an assets directory in the public folder, adding the `logo.png` file, and updating the `index.html` file to include the logo in the header component.

## Changes Made
- Created a new `assets` directory within the `public` folder.
- Added the `logo.png` file to the `public/assets` directory.
- Modified the `index.html` file to integrate the `logo.png` into the header component as a favicon using an `<link>` tag with appropriate styling for proper display.

## Testing
- Verified that the `assets` directory is correctly created in the `public` folder.
- Confirmed that the `logo.png` file is accessible and renders correctly in the browser.
- Tested the header component in `index.html` to ensure the favicon displays as expected across different screen sizes.

Winnycloud27 committed:
feat(assets): create assets directory and add app logo

details:
- created assets directory in public folder
- added app logo file to public/assets

impact:
- enables consistent access to app logo for UI components
- improves asset organization for frontend resources

Winnycloud27 committed:
feat(index.html): integrate app logo into index.html

details:
- added favicon link to index.html using `<link rel="icon" type="image/x-icon" href="./assets/burger-logo.png">`
- references logo file in public/assets directory

impact:
- displays app logo as favicon in browser
- enhances branding consistency in UI

Winnycloud27 committed:
Merge branch 'dev' of https://github.com/Winnycloud27/burger-frontend into 4-asset-management-and-branding-integration

Winnycloud27 committed:
Merged dev to 4-asset

adiozdaniel commented:
<img width="341" height="546" alt="image" src="https://github.com/user-attachments/assets/3317f3fb-f01a-4dc3-bd17-f4982db6871c" />

This should be updated to `In review`

adiozdaniel commented:
<img width="341" height="548" alt="image" src="https://github.com/user-attachments/assets/1dfc4dd6-8d5a-4367-b3a3-0f63980b1c23" />


You should also attach it to an issue!

adiozdaniel commented:
Handle the proposed changes!

adiozdaniel commented on src/index.html:
```diff
@@ -2,10 +2,10 @@
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="icon" type="image/x-icon" href="./assets/burger-logo.png">
```
Images should be added in public directory

In public directory, you can add different directories

adiozdaniel replied:
## Below is an example of how you can organize images

```markdown
burger-frontend/
├── public/
│   ├── images/
│   │   ├── branding/
│   │   │   ├── logo.svg
│   │   │   ├── logo-white.svg
│   │   │   └── favicon.ico
│   │   ├── heroes/
│   │   │   ├── hero-banner.jpg
│   │   │   └── hero-background.png
│   │   ├── products/                # Product-related images
│   │   ├── users/                   # User avatars, profiles
│   │   │
│   │   └── backgrounds/             # Background patterns, textures
│   │
│   └── documents/
│       ├── terms.pdf
│       └── privacy.pdf
├── src/
│   ├── app/
│   │   ├── core/
│   │   │ 
│   │   ├── features/
│   │   │   ├── products/
│   │   │   │   ├── components/
│   │   │   │   ├── services/
│   │   │   │   ├── models/
│   │   │   │   └── products.module.ts
│   │   │   ├── users/
│   │   │   └── dashboard/
│   │   └── environments/            # Environment configurations
│   └── styles/                      # Global styles
├── angular.json
└── package.json
```

Winnycloud27 committed:
feat(public): reorganize assets directory structure

details:
- Moved burger-logo.png from public/assets/ to public/images/branding/
- changed the favicon name from burger-logo.png to favicon.ico
- Updated references to reflect new file location

impact:
Improves the project's file structure by aligning it with a more logical and scalable directory hierarchy.
May temporarily break existing references to the old path (public/assets/) in the codebase or build configuration, requiring updates to ensure the logo displays correctly.
Enhances maintainability and readability for future development once references are updated.

Winnycloud27 committed:
feat(index.html): integrate favicon into index.html

- Replaced  <link rel="icon" type="image/x-icon" href="./assets/burger-logo.png"> with <link rel="icon" type="image/x-icon" href="./images/branding/favicon.ico"> to index.html
- Ensured favicon is correctly linked from the updated images/branding/ directory

impact:
Ensures the website displays a custom favicon in supported browsers, improving brand consistency.
Has minimal impact on existing functionality, but any misconfiguration in the path or file name could result in the default browser icon being displayed.
Requires proper testing to confirm visibility across different browsers.

Winnycloud27 committed:
feat(public): update file structure

details:
- Removed assets/burger-logo.png from public directory
- Added favicon.ico to public/images/branding/ as the new favicon

impact:
- Ensures maintainability and readability.

Winnycloud27 commented:
> Handle the proposed changes!

I have updated the proposed changes.
